gcloud/storage/blob.py

```diff
@@ -133,6 +134,22 @@ def path(self):
         return self.path_helper(self.bucket.path, self.name)
 
     @property
+    def path_with_params(self):
+        """Getter property for the URL path to this Blob, with version.
+
+        :rtype: tuple of ``path`` (a string) and ``params`` (a dictionary)
+        :returns: the URL path to this blob and a dictionary with the
+                 generation that can be used in query_params for
+                 connection.api_request
+        """
+
+        params = {}
+        if self.generation is not None:
+            params = {'generation': self.generation}
+
+        return (self.path, params)
```

Comment:
Also, I expected this to just be a string, though I'll hold off on comments until I see how it's used below.